src/core_plugins/i18n/server/i18n/i18n.js

```diff
@@ -0,0 +1,130 @@
+import path from 'path';
+import Promise from 'bluebird';
+import { readFile } from 'fs';
+import _ from 'lodash';
+
+const asyncReadFile = Promise.promisify(readFile);
+
+const TRANSLATION_FILE_EXTENSION = '.json';
+
+let registeredTranslations = {};
+
+/**
+ * Return all translations registered for a particular locale.
+ * @param {string} locale - Translation locale to be returned
+ * @return {Promise} - A Promise object which will contain on resolve a JSON object of all registered translations
+ */
+export function getTranslationsForLocale(locale) {
+  if (!registeredTranslations.hasOwnProperty(locale)) {
+    return Promise.resolve({});
+  }
+
+  const translationFiles = registeredTranslations[locale];
+  const translations = _.map(translationFiles, (filename) => {
+    return asyncReadFile(filename, 'utf8')
+    .then(fileContents => JSON.parse(fileContents))
+    .catch(SyntaxError, function (e) {
+      return Promise.reject('Invalid json in ' + filename);
+    }).catch(function (e) {
+      throw new Error('Cannot read file ' + filename);
+    });
+  });
+
+  return Promise.all(translations)
+  .then(translations => _.assign({}, ...translations));
+};
+
+/**
+ * Return all translations registered for the default locale.
+ * @param {Object} server - Hapi server instance
+ * @return {Promise} - A Promise object which will contain on resolve a JSON object of all registered translations
+ */
+export function getTranslationsForDefaultLocale(server) {
+  let defaultLocale = '';
+  try {
+    defaultLocale = server.config.get('i18n.locale');
+  } catch (e) {
+    defaultLocale = 'en';
+  }
+  return getTranslationsForLocale(defaultLocale);
+};
+
+/**
+ * Returns list of all registered locales.
+ * @return {Array} - Registered locales
+ */
+export function getRegisteredTranslationLocales() {
+  return Object.keys(registeredTranslations);
+};
+
+/**
+ * Return translations for a suitable locale from a user side locale list
+ * @param {Sring} languageTags -  BCP 47 language tags
+ * @return {Object} - A Promise object which will contain on resolve a JSON object of all registered translations
+ */
+export function getTranslationsForLocales(languageTags) {
```

Comment:
Can you think of a name that better represents what this function does? I'm having a hard time coming up with something good, but I feel like the current name could be confusing. It sounds like it would return multiple sets of translations for multiple locales.